packages/contracts/contracts/MarginlyPool.sol

```diff
@@ -1441,15 +1468,23 @@ contract MarginlyPool is IMarginlyPool {
 
     if (mode != Mode.Regular) revert Errors.EmergencyMode();
 
-    (bool callerMarginCalled, FP96.FixedPoint memory basePrice) = reinit();
+    address positionOwner = msg.sender;
+    if (positionOwner == IMarginlyFactory(factory).manager()) {
+      if (call != CallType.Long && call != CallType.Short && call != CallType.ClosePosition) {
+        revert Errors.Forbidden();
+      }
+      positionOwner = positionAddress;
+    }
```

Comment:
You could avoid double time assigning `positionOwner` when msg.sender is manager.
```solidity
    address positionOwner;
    if (msg.sender == IMarginlyFactory(factory).manager()) {
      if (call != CallType.Long && call != CallType.Short && call != CallType.ClosePosition) {
        revert Errors.Forbidden();
      }
      positionOwner = positionAddress;
    }
    else{
       positionOwner = msg.sender;
    }
```